### PR TITLE
test: cover missing Purchases functions

### PR DESCRIPTION
**Pull request by vegaro**
Adds `CallbackResponseTests.cs`: customer info, login, logout, restore, sync, offerings, eligibility, storefront, promotional offers, placement offerings, and Amazon LWA consent status.

---

> [!NOTE]
> **Low Risk**
> Test-only additions with no production code changes; risk is limited to test maintenance and documenting known callback edge cases.
> 
> **Overview**
> Adds **`CallbackResponseTests`** edit-mode coverage for the `Purchases` private `_*` receivers that native iOS/Android wrappers invoke via `UnitySendMessage`, using a **`PurchasesWrapperSpy`** and JSON fixtures that mirror hybrid-common wire formats (explicit nulls, error envelopes).
> 
> Tests drive success and error paths for **get products**, **offerings**, **can make payments**, **customer info**, **log in/out**, **restore/sync**, **intro eligibility**, **storefront**, **promotional offers**, **placement offerings**, and **Amazon LWA consent**, including **single-fire callbacks** on duplicate native responses and platform-specific payload shapes.
> 
> One test **documents current behavior**: **`GetCurrentOfferingForPlacement`** treats error-only payloads like “no offering” because the missing-offering guard runs before the error check.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 4edc64aef4164b36f468b8f5e35e9b91bbe1ab77. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by vegaro** [comment]
* **#1012** <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-unity/1012?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-unity/1012?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1009** <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-unity/1009?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 4 other dependent PRs ([#1010](https://github.com/RevenueCat/purchases-unity/pull/1010) <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-unity/1010?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#1011](https://github.com/RevenueCat/purchases-unity/pull/1011) <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-unity/1011?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#1013](https://github.com/RevenueCat/purchases-unity/pull/1013) <a href="https://app.graphite.com/github/pr/RevenueCat/purchases-unity/1013?utm_source=stack-comment-icon" target="_blank">Graphite</a> and 1 other)
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: IntegrationTests/Assets/Tests/EditMode/CallbackResponseTests.cs [hunk 1]
```diff
@@ -0,0 +1,723 @@
...
+        /// explicit null; iOS omits presentedOfferingIdentifier/presentedOfferingContext entirely.
+        private const string NonSubscriptionProductJson =
+            "{\"identifier\":\"lifetime\",\"description\":\"Lifetime access\",\"title\":\"Lifetime\"," +
+            "\"price\":99.99,\"priceString\":\"$99.99\",\"currencyCode\":\"USD\"," +
+            "\"introPrice\":null,\"discounts\":null," +
+            "\"pricePerWeek\":null,\"pricePerMonth\":null,\"pricePerYear\":null," +
+            "\"pricePerWeekString\":null,\"pricePerMonthString\":null,\"pricePerYearString\":null," +
+            "\"productCategory\":\"NON_SUBSCRIPTION\",\"productType\":\"CONSUMABLE\"," +
+            "\"subscriptionPeriod\":null,\"defaultOption\":null,\"subscriptionOptions\":null," +
+            "\"presentedOfferingIdentifier\":null,\"presentedOfferingContext\":null}";
```
Comment (vegaro) [inline IntegrationTests/Assets/Tests/EditMode/CallbackResponseTests.cs:60]: already thinking of ways about how we can derive these from phc directly without hardcoding them here 🤔

File: IntegrationTests/Assets/Tests/EditMode/CallbackResponseTests.cs [hunk 2]
```diff
@@ -0,0 +1,723 @@
...
+using RevenueCat.SimpleJSON;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace RevenueCat.Tests
+{
+    /// <summary>
+    /// Exercises the <c>_*</c> receivers that the native wrappers call through
+    /// <c>UnitySendMessage</c>. Every JSON payload here mirrors what
+    /// purchases-hybrid-common 18.29.0 hands to <c>PurchasesUnityHelper.m</c> /
```
Comment (tonidero) [inline IntegrationTests/Assets/Tests/EditMode/CallbackResponseTests.cs]: Adding the version here seems like it will get outdated really soon 😂

File: IntegrationTests/Assets/Tests/EditMode/CallbackResponseTests.cs [hunk 3]
```diff
@@ -0,0 +1,723 @@
...
+            "\"price\":99.99,\"priceString\":\"$99.99\",\"currencyCode\":\"USD\"," +
+            "\"introPrice\":null,\"discounts\":null," +
+            "\"pricePerWeek\":null,\"pricePerMonth\":null,\"pricePerYear\":null," +
+            "\"pricePerWeekString\":null,\"pricePerMonthString\":null,\"pricePerYearString\":null," +
+            "\"productCategory\":\"NON_SUBSCRIPTION\",\"productType\":\"CONSUMABLE\"," +
+            "\"subscriptionPeriod\":null,\"defaultOption\":null,\"subscriptionOptions\":null," +
+            "\"presentedOfferingIdentifier\":null,\"presentedOfferingContext\":null}";
+
+        /// Offering.map() / Offering.dictionary for an offering with no packages.
+        private const string OfferingJson =
```
Comment (tonidero) [inline IntegrationTests/Assets/Tests/EditMode/CallbackResponseTests.cs]: Hmm maybe `EmptyOfferingJson`?

**Review by tonidero — APPROVED** [review]
Nice!!

File: IntegrationTests/Assets/Tests/EditMode/CallbackResponseTests.cs (hunk 2 above)
Comment (vegaro) [reply IntegrationTests/Assets/Tests/EditMode/CallbackResponseTests.cs]: Dropped the version. It proved the point faster than expected, I had to bump it from 18.25.0 to 18.29.0 while this branch was in review, because rebasing onto main picked up two PHC bumps

File: IntegrationTests/Assets/Tests/EditMode/CallbackResponseTests.cs (hunk 3 above)
Comment (vegaro) [reply IntegrationTests/Assets/Tests/EditMode/CallbackResponseTests.cs]: Renamed, but to `OfferingWithoutPackagesJson` rather than `EmptyOfferingJson`
